sql/src/main/java/io/crate/execution/engine/collect/sources/InformationSchemaIterables.java

```diff
@@ -280,6 +291,43 @@ public ConstraintInfo next() {
         }
     }
 
+    static class CheckConstraintIterable implements Iterable<ConstraintInfo> {
+
+        private final RelationInfo info;
+
+        CheckConstraintIterable(RelationInfo info) {
+            this.info = info;
+        }
+
+        @Override
+        public Iterator<ConstraintInfo> iterator() {
+            return new CheckConstraintIterator(info);
+        }
+    }
+
+    static class CheckConstraintIterator implements Iterator<ConstraintInfo> {
```

Comment:
If we're already working with the Stream API further above, do we even need this? E.g. Could use a `map(x -> new ConstraintInfo(...)`, no?